transformations/unit_converter/transformation.py

```diff
@@ -0,0 +1,144 @@
+        try:
+            q = float(q)
+        except ValueError:
+            q = w2n.word_to_num(
+                q
+            )  # if it is not a numerical value, convert it and remember to convert it back after unit conversion
+            is_word = True
+        q /= converter[
+            plurals[X.root.lemma_]
+        ]  # convert to base unit (meters, kilograms)
+        new_unit = random.choice(
+            units
+        )  # pick a random new unit from the available one (the original one is included)
+        while not not_zero:
+            if new_unit in plurals:
+                new_unit = plurals[new_unit]  # set to plural form
+            q_new = (
+                q * converter[new_unit]
+            )  # convert from base unit to new unit
+            q_new = float("{:.3f}".format(q_new))  # clean format
+            if q_new != 0:
+                not_zero = True
+                q = q_new
+            else:  # it happens if the conversion is not optimal (e.g., we avoid to convert 1 centimeter to kilometers)
+                new_unit = random.choice(units)
+
+        if (
+            int(q) == 1 and q - int(q) < 0.01
+        ):  # if the converted number of units is 1, put to singular
+            new_unit = list(plurals.keys())[
+                list(plurals.values()).index(new_unit)
+            ]
+
+        if (
+            is_word
+        ):  # if the quantity was in word format, put it back to word format
+            q = num2words(q).replace("-", " ")
+        else:
+            q = str(q)
+        new_text = q + " " + new_unit
+        text = text.replace(X.text, new_text)
+    return text
+
+
+def convert_units(self, text, seed=0, max_outputs=1):
+    random.seed(seed)
+    doc = self.nlp(text)
+    perturbed_texts = []
+    for _ in itertools.repeat(None, max_outputs):
+        perturbed_text = text
+        for X in doc.ents:
+            if X.label_ == "QUANTITY":
+                perturbed_text = convert(
+                    perturbed_text,
+                    X,
+                    self.units_lengths,
+                    self.plurals_lengths,
+                    self.converter_lengths,
+                )
+                perturbed_text = convert(
+                    perturbed_text,
+                    X,
+                    self.units_weights,
+                    self.plurals_weights,
+                    self.converter_weights,
+                )
+        perturbed_texts.append(perturbed_text)
+
+    return perturbed_texts
+
+
+class UnitConverter(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en"]
+
+    def __init__(self, seed=0, max_outputs=1):
+        super().__init__(seed, max_outputs=max_outputs)
+        self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+
+        self.converter_lengths = {
+            "miles": 0.00062137121,
```

Comment:
Would it be useful to encode the `converter_lengths`, `converter_weights`, `plurals_lengths` into JSON files so that it is easily extendable in the future?

Reply:
Thank you @timothy22000 for the suggestion. I agree that it could be useful and I have added the dictionaries in a JSON file. ca881471cb20f679cd41d2eecc430cbc8db133ee